src/base.js

```diff
@@ -55,7 +55,7 @@ class EntryList {
           if (comp > -1) {
             found.push(keys.pop())
           } else {
-            break
+            break // this is the only branch that hits if the new key sorts before the earliest key in the tree
           }
```

Comment:
I simplifed the test a little and stepped through on the debugger until I found this. I'm not sure what the English description would be of correct behavior for putting a new front entry on the tree. Maybe we can connect on that @mikeal and then I'm happy to implement